pkg/site/planning.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package site
+
+import (
+	"fmt"
+	"html/template"
+	"net/http"
+	"path/filepath"
+	"strings"
+
+	"github.com/google/triage-party/pkg/hubbub"
+	"github.com/google/triage-party/pkg/provider"
+	"github.com/google/triage-party/pkg/triage"
+	"k8s.io/klog/v2"
+)
+
+const (
+	unplanned  = "Not Planned"
+	inProgress = "In Progress"
+)
+
+// Planning shows a view of a collection.
+func (h *Handlers) Planning() http.HandlerFunc {
+	fmap := template.FuncMap{
+		"toJS":          toJS,
+		"toYAML":        toYAML,
+		"toJSfunc":      toJSfunc,
+		"toDays":        toDays,
+		"HumanDuration": humanDuration,
+		"UnixNano":      unixNano,
+		"getAssignees":  getAssignees,
+		"getMilestone":  getMilestone,
+		"Avatar":        avatarSticky,
+		"Class":         className,
+	}
+
+	t := template.Must(template.New("planning").Funcs(fmap).ParseFiles(
+		filepath.Join(h.baseDir, "planning.tmpl"),
+		filepath.Join(h.baseDir, "base.tmpl"),
+	))
+
+	return func(w http.ResponseWriter, r *http.Request) {
+		klog.Infof("GET %s: %v", r.URL.Path, r.Header)
+
+		id := strings.TrimPrefix(r.URL.Path, "/p/")
+
+		p, err := h.collectionPage(r.Context(), id, isRefresh(r))
+		if err != nil {
+			http.Error(w, fmt.Sprintf("planning page for %q: %v", id, err), 500)
+			klog.Errorf("page: %v", err)
+
+			return
+		}
+
+		if p.CollectionResult.RuleResults != nil {
+			p.Description = p.Collection.Description
+			p.Swimlanes = groupByState(p.CollectionResult.RuleResults)
+		}
+
+		err = t.ExecuteTemplate(w, "base", p)
+
+		if err != nil {
+			klog.Errorf("tmpl: %v", err)
+			return
+		}
+	}
+}
+
+func getMilestone(c *hubbub.Conversation) string {
+	if c.Milestone == nil || len(c.PullRequestRefs) == 0 {
+		return ""
+	}
+	return fmt.Sprintf("Milestone: %s", *c.Milestone.Title)
+}
+
+func avatarSticky(u *provider.User) template.HTML {
```

Comment:
Can this change be moved into the `avatar` function?

If not, please add a comment describing the behavioral differences. It isn't clear to me why this is sometimes wrapping a `div` around versus not.

Reply:
used Avatar function.